docs/source/dev/front-end.md

```diff
@@ -1,9 +1,76 @@
 # Front-end architecture
 
+## Tooling
+
+The front-end package manager is [pnpm](https://pnpm.io/):
+
+- `pnpm install` - install the dependencies, updating the lockfile as needed (except when the `CI` env var is defined)
+- `pnpm add [--save-dev] <pkg-name>` - [add a dependency](https://pnpm.io/cli/add)
+- `pnpm why <pkg-name>` - show why a package is present in `node_modules`
+- `pnpm up -L <pkg-name>` - update a package to the latest version
+- `pnpm outdated` - list outdated dependencies
+- `pnpm [run] <script> [--<arg>]` - run a script defined in `package.json`
+- `pnpm dlx <pkg-name>` - fetch a package from the registry and run its default
+  command binary (equivalent to `npx <pkg-name>`)
+
+> You can run all `pnpm` commands above from the root of the repository with `pnpm --prefix ui <cmd>`.
+
+[Vite](https://vitejs.dev/) is used to build the project for production:
+
+- `pnpm start` - start the development server at http://localhost:5173
+- `pnpm build` - build the front-end for production into the `ui/build` folder
+- `pnpm preview` - serve the production build at http://localhost:5173
+
+The codebase is formatted with Prettier, linted with ESLint and tested with Cypress:
+
+- `pnpm prettier` - check formatting
+- `pnpm format` - fix formatting
+- `pnpm eslint` - check for linting errors
+- `pnpm fix` - fix linting errors
+- `pnpm cypress` - open Cypress testing UI
+- `pnpm e2e` - run Cypress E2E tests
```

Comment:
@axelboc Just noticed... Shouldn't these be `pnpm run <something>`?

Reply:
OK, apparently not... I am surprised

Reply:
Yeah, very convenient. Works with any custom script that is not a known `pnpm` command.